ouroboros-network/src/Ouroboros/Network/TxSubmission/Outbound.hs

```diff
@@ -69,7 +69,13 @@ data MempoolSnapshot txid tx idx =
        mempoolLookupTx   :: idx -> Maybe tx
      }
 
-data TraceTxSubmissionOutbound txid tx = TraceTxSubmissionOutbound --TODO
+data TraceTxSubmissionOutbound txid tx
+  = TraceTxSubmissionOutboundRecvMsgRequestTxs
+      ![txid]
+      -- ^ The IDs of the transactions requested.
+  | TraceTxSubmissionOutboundSendMsgReplyTxs
```

Comment:
@mrBliss: Think my constructor names are too long?

Reply:
They are self-explanatory and seem to still fit within the 80 char limit, so it's fine for me.